src/main/java/com/github/daniilandco/vehicle_sales_project/controller/UserController.java

```diff
@@ -0,0 +1,74 @@
+package com.github.daniilandco.vehicle_sales_project.controller;
+
+import com.github.daniilandco.vehicle_sales_project.controller.request.NewAdRequest;
+import com.github.daniilandco.vehicle_sales_project.controller.request.RegisterRequest;
+import com.github.daniilandco.vehicle_sales_project.service.ad.AdServiceImplementation;
+import com.github.daniilandco.vehicle_sales_project.service.user.UserServiceImplementation;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+
+    private final AdServiceImplementation adServiceImplementation;
+    private final UserServiceImplementation userServiceImplementation;
+
+    public UserController(AdServiceImplementation adServiceImplementation, UserServiceImplementation userServiceImplementation) {
+        this.adServiceImplementation = adServiceImplementation;
+        this.userServiceImplementation = userServiceImplementation;
+    }
+
+    @GetMapping("/ads")
+    public ResponseEntity<?> getUserAds() {
+        return adServiceImplementation.getUserAds();
+    }
+
+    @GetMapping("/ads/{id}")
+    public ResponseEntity<?> getAdById(@PathVariable Long id) {
+        return adServiceImplementation.getUserAdById(id);
+    }
+
+    @PostMapping("/add")
+    public ResponseEntity<?> addAd(@RequestBody NewAdRequest request) {
+        return adServiceImplementation.addAd(request);
+    }
```

Comment:
why not moving ad-related endpoints to `AdController` with a base context `/ads`?